ui/ui-components/pages/model/[modelId]/source/[sourceId]/edit.tsx

```diff
@@ -86,8 +86,8 @@ const Page: NextPage<Props> = ({
   }, [preview]);
 
   const mappingColumn = useMemo(
-    () => Object.keys(source.mapping)[0] ?? previewColumns?.[0],
```

Comment:
I don't recall why this was added in the previews change, so I restored it to the previous version. @krishnaglick do you recall the reason?

Reply:
I believe it needed to be up here so the data was correct in all the spots. Once we do a bigger refactor this will likely go away.

Reply:
Ok I restored it and tested it and still works